packages/tizen_rpc_port/tizen/src/rpc_port_proxy.cc

```diff
@@ -0,0 +1,223 @@
+  int ret = rpc_port_proxy_create(&handle_);
+  if (ret != RPC_PORT_ERROR_NONE)
+    LOG_ERROR("rpc_port_proxy_create() is failed. error: %d", ret);
+}
+
+RpcPortProxy::~RpcPortProxy() {
+  if (handle_ != nullptr) rpc_port_proxy_destroy(handle_);
+}
+
+RpcPortResult RpcPortProxy::Connect(EventSink sink) {
+  LOG_DEBUG("Connect: %s/%s", appid_.c_str(), port_name_.c_str());
+  if (connected_) {
+    LOG_ERROR("Already connected");
+    return CreateResult(RPC_PORT_ERROR_INVALID_PARAMETER);
+  }
+
+  event_sink_ = std::move(sink);
+  int ret =
+      rpc_port_proxy_add_connected_event_cb(handle_, OnConnectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_disconnected_event_cb(handle_, OnDisconnectedEvent,
+                                                 this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_rejected_event_cb(handle_, OnRejectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_received_event_cb(handle_, OnReceivedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_connect(handle_, appid_.c_str(), port_name_.c_str());
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  connected_ = true;
+  return CreateResult(RPC_PORT_ERROR_NONE);
+}
+
+RpcPortResult RpcPortProxy::ProxyConnectSync(EventSink sink) {
+  LOG_DEBUG("ProxyConnectSync: %s/%s", appid_.c_str(), port_name_.c_str());
+  if (connected_) {
+    LOG_ERROR("Already connected");
+    return CreateResult(RPC_PORT_ERROR_INVALID_PARAMETER);
+  }
+
+  event_sink_ = std::move(sink);
+  int ret =
+      rpc_port_proxy_add_connected_event_cb(handle_, OnConnectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_disconnected_event_cb(handle_, OnDisconnectedEvent,
+                                                 this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_rejected_event_cb(handle_, OnRejectedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret = rpc_port_proxy_add_received_event_cb(handle_, OnReceivedEvent, this);
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  ret =
+      rpc_port_proxy_connect_sync(handle_, appid_.c_str(), port_name_.c_str());
+  if (ret != RPC_PORT_ERROR_NONE) {
+    return CreateResult(ret);
+  }
+
+  connected_ = true;
+  return CreateResult(RPC_PORT_ERROR_NONE);
+}
+
+RpcPortResult RpcPortProxy::GetPort(int32_t type, RpcPort** port) {
+  rpc_port_h port_native = nullptr;
+  int ret = rpc_port_proxy_get_port(
+      handle_, static_cast<rpc_port_port_type_e>(type), &port_native);
+  if (ret != RPC_PORT_ERROR_NONE) return CreateResult(ret);
+
+  *port = new RpcPort(port_native, type);
```

Comment:
This may cause memory leak. How about passing `std::unique_ptr<RpcPort>` type and calling `port->reset(new RpcPort(port_native, type));`?

Reply:
I agree with your suggestion.
I apply it. Thanks :)